storage/postgres/query_builder.go

```diff
@@ -0,0 +1,330 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/jmoiron/sqlx"
+)
+
+type orderRule struct {
+	field     string
+	orderType query.OrderType
+}
+
+type QueryBuilder struct {
+	db pgDB
+
+	sql         string
+	queryParams []interface{}
+
+	labelCriteria, fieldCriteria []query.Criterion
+	orderByFields                []orderRule
+	limit                        int
+	criteria                     []query.Criterion
+	hasLock                      bool
+	returningFields              []string
+
+	err error
+}
+
+func NewQueryBuilder(db pgDB) *QueryBuilder {
+	return &QueryBuilder{
+		db: db,
+	}
+}
+
+func (qb *QueryBuilder) List(ctx context.Context, entity PostgresEntity) (*sqlx.Rows, error) {
+	if qb.err != nil {
+		return nil, qb.err
+	}
+
+	baseTableName := entity.TableName()
+	var baseQuery string
+	label := entity.LabelEntity()
+	if label == nil {
```

Comment:
can this be moved to `constructBaseQueryForLabelable` where `if label == nil { return the code from constructBaseQueryForEntity}` and delete `constructBaseQueryForEntity`